commit 3746e209d88137206b5ad319dc049207bc4e2524
Author: John Spray <jspray@redhat.com>
Date:   Thu Mar 30 10:07:23 2017 -0400

    mgr: embed Client instance for mds_command
    
    Now plugins can talk to MDS daemons.
    
    Signed-off-by: John Spray <john.spray@redhat.com>

diff --git a/src/mgr/PyModules.cc b/src/mgr/PyModules.cc
index 7501ac55e29..3e00822fc28 100644
--- a/src/mgr/PyModules.cc
+++ b/src/mgr/PyModules.cc
@@ -61,10 +61,10 @@ namespace {
 #define dout_prefix *_dout << "mgr " << __func__ << " "
 
 PyModules::PyModules(DaemonStateIndex &ds, ClusterState &cs, MonClient &mc,
-                     Objecter &objecter_,
+                     Objecter &objecter_, Client &client_,
 		     Finisher &f)
   : daemon_state(ds), cluster_state(cs), monc(mc),
-    objecter(objecter_),
+    objecter(objecter_), client(client_),
     finisher(f)
 {}
 
diff --git a/src/mgr/PyModules.h b/src/mgr/PyModules.h
index b5fdde8a776..8e929113053 100644
--- a/src/mgr/PyModules.h
+++ b/src/mgr/PyModules.h
@@ -20,6 +20,9 @@
 #include "common/Mutex.h"
 #include "common/Thread.h"
 
+#include "osdc/Objecter.h"
+#include "client/Client.h"
+
 #include "DaemonState.h"
 #include "ClusterState.h"
 
@@ -33,6 +36,7 @@ class PyModules
   ClusterState &cluster_state;
   MonClient &monc;
   Objecter &objecter;
+  Client   &client;
   Finisher &finisher;
 
   mutable Mutex lock{"PyModules"};
@@ -43,7 +47,7 @@ public:
   static constexpr auto config_prefix = "mgr.";
 
   PyModules(DaemonStateIndex &ds, ClusterState &cs, MonClient &mc,
-            Objecter &objecter_,
+            Objecter &objecter_, Client &client_,
             Finisher &f);
 
   ~PyModules();
@@ -51,6 +55,7 @@ public:
   // FIXME: wrap for send_command?
   MonClient &get_monc() {return monc;}
   Objecter  &get_objecter() {return objecter;}
+  Client    &get_client() {return client;}
 
   PyObject *get_python(const std::string &what);
   PyObject *get_server_python(const std::string &hostname);
